src/util/CancellationToken.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <atomic>
+#include <memory>
+
+namespace chatterino {
+
+class CancellationToken
+{
+public:
+    CancellationToken() = default;
+    explicit CancellationToken(bool isCanceled)
+        : isCanceled_(new std::atomic<bool>(isCanceled))
+    {
+    }
+
+    CancellationToken(const CancellationToken &) = default;
+    CancellationToken(CancellationToken &&other)
+        : isCanceled_(std::move(other.isCanceled_)){};
+
+    CancellationToken &operator=(CancellationToken &&other)
+    {
+        this->isCanceled_ = std::move(other.isCanceled_);
+        return *this;
+    }
+    CancellationToken &operator=(const CancellationToken &) = default;
+
+    void cancel()
+    {
+        if (this->isCanceled_ != nullptr)
+        {
+            this->isCanceled_->store(true, std::memory_order_release);
+        }
+    }
+
+    bool isCanceled() const
```

Comment:
Should be `isCancelled` to be consistent with US/Bri'ish spelling - same goes for the variable